### PR TITLE
feat: increase GLOO timeout [DET-3309]

### PR DESCRIPTION
**Pull request by aaron276h**
## Description
When running on a large number of nodes (40+), the default timeout of 30 seconds is not enough for all nodes to connect with the chief process.

## Test Plan
Tested by spinning up a 48 node cluster and successfully running a training job.

### CODE REVIEW

File: harness/determined/horovod.py [hunk 1]
```diff
@@ -143,6 +143,8 @@ def create_run_command(
         create_hostlist_arg(num_gpus_per_machine, ip_addresses),
         "--start-timeout",
         str(constants.HOROVOD_STARTUP_TIMEOUT_SECONDS),
+        "--gloo-timeout-seconds",
+        str(constants.HOROVOD_GLOO_TIMEOUT_SECONDS),
```
Comment (brain-good) [inline harness/determined/horovod.py:147]: idea: wonder if we should actually do some scaling by the number of instances..

Comment (aaron276h) [reply harness/determined/horovod.py:147]: IMO that would be unnecessary complexity. There really isn't any downside to having a 4-minute timeout (famous last words) for this.

Comment (brain-good) [reply harness/determined/horovod.py:147]: I was more thinking of if we go even more instances, are we going to have to keep bumping this. But I'm fine either way

Comment (aaron276h) [reply harness/determined/horovod.py:147]: That's a good question. 4-minutes should already give a lot of space to scale up so I'm inclined to handle that case if/when we hit in the future.

**Pull request merged**